custom_metrics/almanac.js

```diff
@@ -306,6 +359,240 @@ return JSON.stringify({
       return null;
     }
   })(),
-  //  check if there is any picture tag containing an img tag
-  'has_picture_img': document.querySelectorAll('picture img').length > 0
+
+  // Previously for 04_04.sql in 2019
+  'inline_svg_stats': (() => {
+    const svg_elements = [...document.querySelectorAll('svg')];
+
+    return {
+      total: svg_elements.length,
+      content_lengths: svg_elements.map(svg => svg.outerHTML.length),
+      props: parseNodes(svg_elements),
+    };
+  })(),
+
+  // Various stats of img, source and picture elements
+  'images': (() => {
+    const pictures = document.querySelectorAll('picture');
+    const imgs = [...document.querySelectorAll('img')];
+    const sources = document.querySelectorAll('source');
+    const pictures_with_img = document.querySelectorAll('picture img');
+
+    const images_with_srcset = document.querySelectorAll('img[srcset], source[srcset]');
+    const images_with_sizes = [...document.querySelectorAll('img[sizes], source[sizes]')];
+    const images_using_loading_prop = [...document.querySelectorAll('img[loading], source[loading]')];
+
+    // NOTE: -1 is used to represent images with no alt tag at all. Empty alt tags have a value of 0
+    const alt_tag_lengths = imgs.map(img => {
+      if (!img.hasAttribute('alt')) {
+        return -1;
+      }
+
+      // alt=" " is less correct but comparable to alt="" so we use trim
+      // Also remove duplicate spaces to get a feel for how long alt tags really are
+      return img.alt.trim().replace(/\s+/g, ' ').length;
+    });
+
+    return {
+      total_pictures: pictures.length,
+      total_img: imgs.length,
+      total_sources: sources.length,
+
+      total_with_srcset: images_with_srcset.length,
+      total_with_sizes: images_with_sizes.length,
+      total_pictures_with_img: pictures_with_img.length,
+
+      // Values specific properties. Cleaned and trimmed to make processing easier
+      sizes_values: images_with_sizes.map(img => {
+        const value = img.getAttribute('sizes') || '';
+        return value.replace(/[\r\n]+/g, ' ').replace(/\s+/g, ' ').trim();
+      }),
+      loading_values: images_using_loading_prop.map(img => {
+        const value = img.getAttribute('loading') || '';
+        return value.replace(/[\r\n]+/g, ' ').replace(/\s+/g, ' ').trim();
+      }),
+      alt_tag_lengths: alt_tag_lengths,
+
+      // NOTE: props starting with __ are not actual properties. They were added by this script
+      picture_props: parseNodes(pictures),
+      img_props: imgs.map(img => {
```

Comment:
I now only dump the custom properties images, sources and pictures can have. Since those would be the only additional properties we might want to look at